controls.py

```diff
@@ -0,0 +1,95 @@
+from testit import *
+import db
+
+
+
+def testTables():
+
+    s = db.get_session()
+
+    # Check Creation
+    A = Account(username = 'JOE',password='ultimatej03',email = 'Joeyjoe@joesph.com',account_token = '555XTZ')
+    S = Student()
+    D = Donor()
+    R = Request(requested_by = 4, amount_needed = 60.00, description = "I NEED A TOAD. TRUST ME.")
+    DM = Donation(to_request= 3, from_donor= 4,amount_given=55.9,charge_token='kEYKey', filled=False)
+
+    s.add(A)
+    s.add(S)
+    s.add(D)
+    s.add(R)
+    s.add(DM) 
+    s.commit()
+
+    print(A, A.__table__, A.id)
+    print(S, S.__table__)
+    print(D, D.__table__)
+    print(R, R.__table__)
+    print(DM, DM.__table__)
+
+    s.close()
+
+
+def createStudent(username, password, email):
+    s = db.get_session()
+
+    Acc =  Account(username = username, password=password, email = email)
+    s.add(Acc)
+    s.flush()
+
+    Stu = Student(related_account=Acc.id)
+    s.add(Stu)
+    s.commit()
+    s.close()
+
+
+def createDonor(username, password, email):
+    s = db.get_session()
+
+    Acc =  Account(username = username, password=password, email = email)
+    s.add(Acc)
+    s.flush()
+
+    Don = Donor(related_account=Acc.id)
+    s.add(Don)
+    s.commit()
+    s.close()
+
+
+def openRequest(student_id, amount, reason):
+    s = db.get_session()
+
+    req = Request(requested_by=student_id, amount_needed=amount, description=reason)
+    s.add(req)
+    s.commit()
+    s.close()   
+ 
+
+def donate(request_id, donor_id, amount):
+
+    s = db.get_session()
+    D = s.query(Donor).get(donor_id)

```

Comment:
False, donors have their own personal id so that we can acces them independently of the accounts table, id in these cases will always just be keys for quick access through tables. So line 71 is just grabbing the person who wants to donate by reference of their key number. Did that help?